// rt-overflow-tooltip/esm2022/lib/rt-overflow-tooltip.module.mjs
import { NgModule } from '@angular/core';
import { MatTooltipModule } from '@angular/material/tooltip';
import { RtDefineStrokeWidthModule } from 'rt-define-stroke-width';
import { RtPlatformModule } from 'rt-platform';
import { RtOverflowTooltipDirective } from './rt-overflow-tooltip.directive';
import * as i0 from "@angular/core";
class RtOverflowTooltipModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtOverflowTooltipModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.2", ngImport: i0, type: RtOverflowTooltipModule, declarations: [RtOverflowTooltipDirective], imports: [RtDefineStrokeWidthModule,
            MatTooltipModule,
            RtPlatformModule], exports: [RtOverflowTooltipDirective] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtOverflowTooltipModule, imports: [RtDefineStrokeWidthModule,
            MatTooltipModule,
            RtPlatformModule] });
}
export { RtOverflowTooltipModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtOverflowTooltipModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [RtOverflowTooltipDirective],
                    imports: [
                        RtDefineStrokeWidthModule,
                        MatTooltipModule,
                        RtPlatformModule,
                    ],
                    exports: [RtOverflowTooltipDirective],
                }]
        }] });
