// rt-define-stroke-width/fesm2022/rt-define-stroke-width.mjs
import * as i0 from '@angular/core';
import { Injectable, Inject, NgModule } from '@angular/core';
import { DOCUMENT } from '@angular/common';

class RtDefineStrokeWidthService {
    document;
    /** Canvas context used to measure text width. */
    canvasContext;
    constructor(document) {
        this.document = document;
        const canvas = document.createElement('canvas');
        this.canvasContext = canvas.getContext('2d');
    }
    /**
     * Returns stroke width for the provided text with specified CSS rule.
     */
    getTextWidth(stroke, font) {
        this.canvasContext.font = font;
        const metrics = this.canvasContext.measureText(stroke);
        return metrics.width;
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtDefineStrokeWidthService, deps: [{ token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtDefineStrokeWidthService });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtDefineStrokeWidthService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; } });

class RtDefineStrokeWidthModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtDefineStrokeWidthModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.1", ngImport: i0, type: RtDefineStrokeWidthModule });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtDefineStrokeWidthModule, providers: [RtDefineStrokeWidthService] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtDefineStrokeWidthModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [RtDefineStrokeWidthService],
                }]
        }] });

/*
 * Public API Surface of rt-define-stroke-width
 */

/**
 * Generated bundle index. Do not edit.
 */

export { RtDefineStrokeWidthModule, RtDefineStrokeWidthService };
