// rt-auto-focus/esm2022/lib/rt-autofocus-input.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RtAutofocusInputDirective } from './rt-autofocus-input.directive';
import * as i0 from "@angular/core";
class RtAutofocusInputModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputModule, declarations: [RtAutofocusInputDirective], imports: [CommonModule], exports: [RtAutofocusInputDirective] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputModule, imports: [CommonModule] });
}
export { RtAutofocusInputModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [RtAutofocusInputDirective],
                    imports: [CommonModule],
                    exports: [RtAutofocusInputDirective],
                }]
        }] });
