// rt-auto-focus/fesm2022/rt-auto-focus.mjs
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

class RtAutofocusInputDirective {
    host;
    constructor(host) {
        this.host = host;
    }
    ngAfterViewInit() {
        this.host.nativeElement.focus();
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
    static ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.1", type: RtAutofocusInputDirective, selector: "[rtAutofocus]", ngImport: i0 });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[rtAutofocus]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; } });

class RtAutofocusInputModule {
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputModule, declarations: [RtAutofocusInputDirective], imports: [CommonModule], exports: [RtAutofocusInputDirective] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputModule, imports: [CommonModule] });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtAutofocusInputModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [RtAutofocusInputDirective],
                    imports: [CommonModule],
                    exports: [RtAutofocusInputDirective],
                }]
        }] });

/*
 * Public API Surface of rt-auto-focus
 */

/**
 * Generated bundle index. Do not edit.
 */

export { RtAutofocusInputDirective, RtAutofocusInputModule };
