// rt-platform/esm2022/lib/rt-platform.service.mjs
import { isPlatformBrowser, isPlatformServer } from '@angular/common';
import { Inject, Injectable, Optional, PLATFORM_ID } from '@angular/core';
import { defaultConfig, RtPlatformConfigToken } from './symbols';
import * as i0 from "@angular/core";
class RtPlatformService {
    platformId;
    config;
    constructor(platformId, injectedConfig) {
        this.platformId = platformId;
        /** merge with default config & save in local variable (so it will be never undefined) */
        this.config = Object.assign(defaultConfig, injectedConfig);
    }
    get baseUrl() {
        /** local base url should have higher priority, as build may be run on a local machine */
        if (this.config.localBaseUrl) {
            return this.config.localBaseUrl;
        }
        if (this.config.serverBaseUrl) {
            return this.config.serverBaseUrl;
        }
        return '';
    }
    get isBrowser() {
        return isPlatformBrowser(this.platformId);
    }
    get isServer() {
        return isPlatformServer(this.platformId);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtPlatformService, deps: [{ token: PLATFORM_ID }, { token: RtPlatformConfigToken, optional: true }], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtPlatformService, providedIn: 'root' });
}
export { RtPlatformService };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtPlatformService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }, { type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [RtPlatformConfigToken]
                }] }]; } });
