// rt-platform/esm2022/lib/rt-platform.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RtPlatformService } from './rt-platform.service';
import { factoryFn, RtPlatformConfigToken } from './symbols';
import { WINDOW, WindowService } from './window.service';
import * as i0 from "@angular/core";
class RtPlatformModule {
    static forRoot(config) {
        return {
            ngModule: RtPlatformModule,
            providers: [
                RtPlatformService,
                WindowService,
                {
                    provide: WINDOW,
                    useFactory: factoryFn,
                    deps: [WindowService],
                },
                {
                    provide: RtPlatformConfigToken,
                    useValue: config,
                },
            ],
        };
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtPlatformModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
    static ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.2", ngImport: i0, type: RtPlatformModule, imports: [CommonModule] });
    static ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtPlatformModule, providers: [RtPlatformService, WindowService, { provide: WINDOW, useFactory: factoryFn, deps: [WindowService] }], imports: [CommonModule] });
}
export { RtPlatformModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.2", ngImport: i0, type: RtPlatformModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    imports: [CommonModule],
                    providers: [RtPlatformService, WindowService, { provide: WINDOW, useFactory: factoryFn, deps: [WindowService] }],
                }]
        }] });
