// rt-platform/esm2022/lib/rt-transfer-state.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./rt-platform.service";
import * as i2 from "@angular/platform-browser";
class RtTransferStateService {
    platform;
    transfer;
    constructor(platform, transfer) {
        this.platform = platform;
        this.transfer = transfer;
    }
    getKey(name) {
        return `state-${name}`;
    }
    setState(name, state) {
        const key = this.getKey(name);
        this.transfer.set(key, state);
    }
    getState(name) {
        const key = this.getKey(name);
        return this.transfer.get(key, null);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtTransferStateService, deps: [{ token: i1.RtPlatformService }, { token: i2.TransferState }], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtTransferStateService, providedIn: 'root' });
}
export { RtTransferStateService };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: RtTransferStateService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.RtPlatformService }, { type: i2.TransferState }]; } });
