// rt-platform/esm2022/lib/window.service.mjs
import { isPlatformBrowser } from '@angular/common';
import { Inject, Injectable, InjectionToken, PLATFORM_ID } from '@angular/core';
import * as i0 from "@angular/core";
class WindowService {
    platformId;
    injector;
    _window;
    constructor(platformId, injector) {
        this.platformId = platformId;
        this.injector = injector;
        if (!isPlatformBrowser(platformId)) {
            this._window = { navigator: { userAgent: 'fakeAgent' } };
        }
        else {
            this._window = window;
        }
    }
    get nativeWindow() {
        return this._window;
    }
    calculateHeight() {
        if (isPlatformBrowser(this.platformId)) {
            const appHeight = () => {
                const doc = document.documentElement;
                doc.style.setProperty(`--app-height`, `${window.innerHeight}px`);
            };
            window.addEventListener(`resize`, appHeight);
            appHeight();
            return window.innerHeight;
        }
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: WindowService, deps: [{ token: PLATFORM_ID }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: WindowService });
}
export { WindowService };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.1", ngImport: i0, type: WindowService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }, { type: i0.Injector }]; } });
export const WINDOW = new InjectionToken('ng-toolkit-window');
